lib/src/main/java/space/taran/arklib/domain/preview/PlainPreviewStorage.kt

```diff
@@ -14,29 +21,38 @@ import kotlin.io.path.createDirectories
 import kotlin.io.path.deleteIfExists
 import kotlin.io.path.isDirectory
 
-class PlainPreviewStorage(val root: Path) : PreviewStorage {
+class PlainPreviewStorage(
+    val root: Path,
+    private val index: PlainResourcesIndex,
```

Comment:
Can we inject just flow instead of whole index? I think, we should add the flow into `ResourcesIndex` interface. Then we would inject the flow from `AggregatedResourceIndex` into `AggregatedPreviewStorage`.

The only method of this class which handles data from the flow is `override suspend fun store(path: Path, meta: ResourceMeta)`. `AggregatedPreviewStorage` should find necessary shard for each diff item and call `shard.store(path, meta)`. `PlainPreviewStorage` should implement flow handling as well, but it won't be used in aggregated mode.

And `ResourceMeta` in `store(...)` can be replaced by `ResourceId` now.

Reply:
I think there is nothing wrong with fact that `PlainPreviewStorage` has a link to `PlainResourcesIndex` and listens only to it. It's simple and clear
If we only have flow from `AggregatedResourceIndex`, then we will have to somehow distinguish from which `PlainResourcesIndex` it came and to which `PlainPreviewStorage` it belongs

Reply:
I'm not sure that we really need to distinguish `Plain` and `Aggregate` indexes, but we can merge it for now.